### PR TITLE
avoid unsafe calls to acos by fixing potential floating-point errors …

### PR DESCRIPTION
**Pull request by copyme**
Passing raw data into trigonometrical functions from the standard library is never a good thing to do.

### CODE REVIEW

File: include/igl/copyleft/comiso/nrosy.cpp [hunk 1]
```diff
@@ -803,8 +805,16 @@ Eigen::VectorXd igl::copyleft::comiso::NRosyField::angleDefect()
     {
       Eigen::VectorXd a = V.row(F(i,(j+1)%3)) - V.row(F(i,j));
       Eigen::VectorXd b = V.row(F(i,(j+2)%3)) - V.row(F(i,j));
-      double t = a.transpose()*b;
-      t /= (a.norm() * b.norm());
+      double t = a.transpose() * b;
+      double norm_prod = a.norm() * b.norm();
+      if ( norm_prod > 0. )
```
Comment (jdumas) [inline include/igl/copyleft/comiso/nrosy.cpp]: Hi. Can you remove the space inside the parentheses? `if (norm_prod > 0.)` (same for the rest of the PR, and for the throw)

Comment (copyme) [reply include/igl/copyleft/comiso/nrosy.cpp]: Right. I have to check the coding style later on. Sorry.

File: include/igl/copyleft/comiso/nrosy.cpp [hunk 2]
```diff
@@ -803,8 +805,16 @@ Eigen::VectorXd igl::copyleft::comiso::NRosyField::angleDefect()
...
       Eigen::VectorXd a = V.row(F(i,(j+1)%3)) - V.row(F(i,j));
       Eigen::VectorXd b = V.row(F(i,(j+2)%3)) - V.row(F(i,j));
-      double t = a.transpose()*b;
-      t /= (a.norm() * b.norm());
+      double t = a.transpose() * b;
+      double norm_prod = a.norm() * b.norm();
+      if (norm_prod > 0.)
+        t /= norm_prod;
+      else
+        throw std::runtime_error ("Error in 'igl::copyleft::comiso::NRosyField::angleDefect': division by zero!");
```
Comment (jdumas) [inline include/igl/copyleft/comiso/nrosy.cpp]: Sorry this one also should be `std::runtime_error(...`. Thanks!

Comment (copyme) [reply include/igl/copyleft/comiso/nrosy.cpp]: Sorry, my bad. Anyway fixed now.

**Pull request merged**